packages/apex-node/test/logs/logService.test.ts

```diff
@@ -310,4 +311,77 @@ describe('Apex Log Service Tests', () => {
       expect(queryStub.calledWith(query)).to.be.true;
     });
   });
+
+  describe('prepareTraceFlag', async () => {
+    it('creates a trace flag when there is a default debugLevel', async () => {
+      const debugLevel: QueryResult = {
+        records: [
+          {
+            Id: '7dl5tgg0005PGdTnFUL'
+          }
+        ]
+      };
+      const queryStub = sandboxStub
+        .stub(mockConnection.tooling, 'query')
+        //@ts-ignore
+        .resolves(debugLevel);
+      const createStub = sandboxStub
+        .stub(mockConnection.tooling, 'create')
+        //@ts-ignore
+        .resolves();
+
+      const logService = new LogService(mockConnection);
+      await logService.createTraceFlag('005000000000000abc');
+      assert(queryStub.calledOnce);
+      assert(createStub.calledOnce);
+      //todo: test what's created - additional stub/callFake?
+      //https://github.com/salesforcecli/toolbelt/pull/64/files
+    });
+
+    it('throws an error when creating a trace flag if there is no default debugLevel', async () => {
+      const logService = new LogService(mockConnection);
+      try {
+        await logService.createTraceFlag('005000000000000abc');
+        assert.fail();
+      } catch (e) {
+        expect(e.message).to.include('Debug Level not found');
+      }
+    });
+
+    // it('returns a username for a given user ID', async () => {
+    //   const userResult = { Id: '0035tgg0005PGdTnXUI' };
+    //   //you can't stub out this way silly
```

Comment:
Need to figure out how to stub this out, something for tomorrow: https://github.com/forcedotcom/sfdx-core/blob/f0d469a5/src/connection.ts#L468